workflow re-organization

chunhuazhou opened the pull request:
## DESCRIPTION OF CHANGES: 
This PR includes some re-organization of the workflow, to make it more readable, mainly including the following changes:
- moving data pre-processing to the start of the cycle
- separate the spinup and prod cycle
- re-order the EnKF related tasks to make the workflow more readable
- add control whether or not to run post processing for spinup cycle and prod cycle, respectively
- enable a separate workflow for doing ensemble forecast (DO_ENSFCST) only, aside from the ensemble data assimilation in a separate workflow 

## TESTS CONDUCTED: 
Tests are being conducted on Hera and WCOSS. 

## DEPENDENCIES:

## DOCUMENTATION:

## ISSUE (optional): 

## CONTRIBUTORS (optional): 
Co-authored by @hu5970

chunhuazhou committed:
Move data preprocessing to the start of each cycle
Separate spinup and prod cycle tasks

chunhuazhou committed:
Re-order EnKF related tasks

chunhuazhou committed:
Restore access for exregional_run_analysis.sh

chunhuazhou committed:
Turn off DA for ensfcst runs

chunhuazhou committed:
Add control for post-processing

hu5970 reviewed (commented):
Right now, the system cannot setup ensemble forecast, right?

chunhuazhou commented:
> Right now, the system cannot setup ensemble forecast, right?

We need to make some changes to the ensemble DA workflow first - prep_cyc and the task to save DA analysis.

Commit:
updates for wcoss2:
1 remove gzip
2 change to use gfs 16.3
3 change workflow to set resources in head of xml.

chunhuazhou committed:
Merge pull request #4 from hu5970/ensfcst_1202_ming

updates for wcoss2:
